google/cloud/datastore/client.py

```diff
@@ -199,8 +199,18 @@ def _extended_lookup(
 
     results = []
 
+    transaction_id = None
+    new_transaction_options = None
+    if transaction is not None:
+        transaction_id = transaction.id
+        if transaction._begin_later and transaction._status == transaction._INITIAL:
+            # if transaction hasn't been initialized, initialize it as part of this request
+            new_transaction_options = transaction._options
```

Comment:
This code fragment is repeated multiple times. Do you think we could pull it into a function that returns new_transaction_options? Would that be a good idea to reduce the amount of repeated code?

Maybe something similar for transaction_id too?

Reply:
Sure, I added a new helper